### PR TITLE
Support running FIPS tests with Oracle JVM

### LINKED ISSUE #66821: 1.2.840.113549.1.1.4 Signature not available for BCJSSE in fips mode
**Build scan**:
https://gradle-enterprise.elastic.co/s/sl7mn5m3gqtts/console-log?task=:client:sniffer:test

**Repro line**:

**Reproduces locally?**:

**Applicable branches**:

**Failure history**:

**Failure excerpt**:

```
: Test class org.elasticsearch.client.sniff.SniffOnFailureListenerTests |  
-- | --
  | 2> Dec 26, 2020 5:01:56 AM org.bouncycastle.jsse.provider.ProvKeyManagerFactorySpi getDefaultKeyStore |  
  | 2> INFO: Initializing empty key store |  
  | 2> Dec 26, 2020 5:01:56 AM org.bouncycastle.jsse.provider.ProvTrustManagerFactorySpi getDefaultTrustStore |  
  | 2> INFO: Initializing with trust store at path: /dev/shm/elastic+elasticsearch+master+matrix-java-periodic-fips/ES_RUNTIME_JAVA/java11/nodes/general-purpose/client/sniffer/build/fips-resources/cacerts.bcfks |  
  | 2> java.lang.ExceptionInInitializerError |  
  | at __randomizedtesting.SeedInfo.seed([E9613EC44B53D6F:A6FD937FBFBB3E3D]:0) |  
  | at java.base/javax.crypto.ProviderVerifier.verify(ProviderVerifier.java:128) |  
  | at java.base/javax.crypto.JceSecurity.verifyProvider(JceSecurity.java:191) |  
  | at java.base/javax.crypto.JceSecurity.getVerificationResult(JceSecurity.java:217) |  
  | at java.base/javax.crypto.JceSecurity.getInstance(JceSecurity.java:141) |  
  | at java.base/javax.crypto.Mac.getInstance(Mac.java:273) |  
  | at org.bouncycastle.jcajce.provider.ProvBCFKS$BCFIPSKeyStoreSpi.calculateMac(Unknown Source) |  
  | at org.bouncycastle.jcajce.provider.ProvBCFKS$BCFIPSKeyStoreSpi.verifyMac(Unknown Source) |  
  | at org.bouncycastle.jcajce.provider.ProvBCFKS$BCFIPSKeyStoreSpi.engineLoad(Unknown Source) |  
  | at java.base/java.security.KeyStore.load(KeyStore.java:1479) |  
  | at org.bouncycastle.jsse.provider.ProvTrustManagerFactorySpi.getDefaultTrustStore(ProvTrustManagerFactorySpi.java:98) |  
  | at org.bouncycastle.jsse.provider.ProvSSLContextSpi.getDefaultTrustManagers(ProvSSLContextSpi.java:302) |  
  | at org.bouncycastle.jsse.provider.DefaultSSLContextSpi$LazyManagers.<clinit>(DefaultSSLContextSpi.java:57) |  
  | at org.bouncycastle.jsse.provider.DefaultSSLContextSpi.<init>(DefaultSSLContextSpi.java:86) |  
  | at org.bouncycastle.jsse.provider.BouncyCastleJsseProvider$7.createInstance(BouncyCastleJsseProvider.java:217) |  
  | at org.bouncycastle.jsse.provider.BouncyCastleJsseProvider$BcJsseService.newInstance(BouncyCastleJsseProvider.java:373) |  
  | at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:236) |  
  | at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:164) |  
  | at java.base/javax.net.ssl.SSLContext.getInstance(SSLContext.java:168) |  
  | at java.base/javax.net.ssl.SSLContext.getDefault(SSLContext.java:99) |  
  | at org.elasticsearch.client.RestClientBuilder.createHttpClient(RestClientBuilder.java:222) |  
  | at java.base/java.security.AccessController.doPrivileged(Native Method) |  
  | at org.elasticsearch.client.RestClientBuilder.build(RestClientBuilder.java:201) |  
  | at org.elasticsearch.client.sniff.SniffOnFailureListenerTests.testSetSniffer(SniffOnFailureListenerTests.java:49) |  
  |   |  
  | Caused by: |  
  | java.lang.SecurityException: Framework jar verification can not be initialized |  
  | at java.base/javax.crypto.JarVerifier.<clinit>(JarVerifier.java:198) |  
  | ... 23 more |  
  |   |  
  | Caused by: |  
  | java.security.PrivilegedActionException: java.security.NoSuchAlgorithmException: 1.2.840.113549.1.1.4 Signature not available |  
  | at java.base/java.security.AccessController.doPrivileged(Native Method) |  
  | at java.base/javax.crypto.JarVerifier.<clinit>(JarVerifier.java:166) |  
  | ... 23 more |  
  |   |  
  | Caused by: |  
  | java.security.NoSuchAlgorithmException: 1.2.840.113549.1.1.4 Signature not available |  
  | at java.base/java.security.Signature.getInstance(Signature.java:267) |  
  | at org.bouncycastle.jcajce.provider.X509CertificateObject.getSignatureFromProvider(Unknown Source) |  
  | at org.bouncycastle.jcajce.provider.X509CertificateObject.verify(Unknown Source) |  
  | at java.base/javax.crypto.JarVerifier.testSignatures(JarVerifier.java:735) |  
  | at java.base/javax.crypto.JarVerifier$1.run(JarVerifier.java:192) |  
  | at java.base/javax.crypto.JarVerifier$1.run(JarVerifier.java:167) |  
  | ... 25 more

```

It's not obvious what triggers this and it is only occured in certain tests and not always or when starting elasticsearch in fips mode. I presume it has to do with a combination of jdk provider and version.  @BigPandaToo has been seeing this locally too IIRC

It looks like we are hitting https://bugs.openjdk.java.net/browse/JDK-8198743 and the appropriate workaround would be to add the SunRsaSign provider too in the list of providers we pack for CI in fips_java.security file, but we should look into why this is not always triggered and the verification of what jar causes this ( I.e. where is the md5withrsa signature used)

### PR DESCRIPTION
**Pull request by ywangd**
This PR conditionally allow `SunRsaSign` to be used as a security provider when the runtime java is an implementation by Oracle. It is necessary because:
* Oracle JVM forces Security Provider verification 
* The verification class (`javax.crypto.JarVerifier`) uses hardcoded certificates with md5WithRsa signature.

Resolves: #66821

### CODE REVIEW

File: gradle/fips.gradle [hunk 1]
```diff
@@ -5,9 +5,11 @@ import org.elasticsearch.gradle.testclusters.TestDistribution
 // Common config when running with a FIPS-140 runtime JVM
 if (BuildParams.inFipsJvm) {
 
+  String javaSecurityFilename = BuildParams.runtimeJavaDetails.startsWith('Oracle') ? 'fips_oracle_java.security' : 'fips_java.security'
```
Comment (ywangd) [inline gradle/fips.gradle:8]: I took a simpler approach that duplicates the `.security` file. It should also be possible to add a line of ` # security.provider.5=SunRsaSign` into the existing file and uncomment it if the runtime java is oracle. But my gradlefu is not great enough to tell exactly where this should fit. Suggestions are welcome.

Comment (jkakavas) [reply gradle/fips.gradle:8]: We already do something similar in 7.x for Zulu 8, I think it s  fine

Comment (jkakavas) [inline gradle/fips.gradle:8]: Is the behavior / code of jar verifier the same in all oracle jdks > 11 ?

**Review by jkakavas — APPROVED** [review]
Lgtm if this covers all versions > 11, thanks Yang !

File: gradle/fips.gradle (hunk 1 above)
Comment (ywangd) [reply gradle/fips.gradle]: Yep it's the same.

**Pull request merged**